packages/build-utils/src/lambda.ts

```diff
@@ -47,19 +55,18 @@ export class Lambda {
    */
   zipBuffer?: Buffer;
 
-  constructor({
-    files,
-    handler,
-    runtime,
-    maxDuration,
-    memory,
-    environment = {},
-    allowQuery,
-    regions,
-    zipBuffer,
-  }: LambdaOptions) {
-    if (!zipBuffer) {
-      assert(typeof files === 'object', '"files" must be an object');
+  constructor(opts: LambdaOptions) {
+    const {
+      handler,
+      runtime,
+      maxDuration,
+      memory,
+      environment = {},
+      allowQuery,
+      regions,
+    } = opts;
+    if ('files' in opts) {
+      assert(typeof opts.files === 'object', '"files" must be an object');
     }
```

Comment:
Should we also check zipBuffer too if it exists?

Reply:
Added.